### DISCUSSION TITLE
Poll BGP session attributes from routers

### DISCUSSION THREAD

**Discussion by daverdzx** [peering-manager/peering-manager#618] [Q&A] [answered]
Hello Guys, I am new to PM.  Please can you confirm if there is a button, a script, job or something that given a set of routers automatically connects to the routers, and extracts all the BGP attributes ( local asn, peer asn, local ip, peer ip, group, import/export policies) to automatically populate the "Direct Peering Sessions" tab inside the "Devices > Routers" module for each router.

Thank you!!

![image](https://user-images.githubusercontent.com/111540778/185507269-9d68226a-3b33-44ee-91ca-6698532a3c7d.png)

**Comment by daverdzx** [comment]
By the way I tried the "Poll Sessions" green button but if there is nothing configured on the "Direct Peering Sessions" Tab, it sends the following error: "Failure polling BGP sessions state" as shown in the image, until I configure some sessions manually then the "Poll Sessions" button gets the state of each session (Active/Established ... etc).  But what I was hoping for the "Poll Sessions" button to do is not only to measure the state of the manually configured sessions in the "Direct Peering Sessions" Tab, but also to configure them automatically and fill all the needed data such as as ASNs/groups/policies etc. 

Please is there a way, a button, or a function to do that?, or maybe my installation is not ok and my "Poll Sessions" button is not working as it should?

Thank you for your help.

![image](https://user-images.githubusercontent.com/111540778/185509476-508c924d-e221-4ba9-a769-64adef55c2df.png)

**Comment by gmazoyer** [comment, marked as answer]
Direct BGP sessions cannot be imported from routers because Peering Manager has standard way to determine the relationships for sessions. The import feature is limited to sessions setup over IXPs.

To make sure the NAPALM credentials are correct and that Peering Manager can connect to your router, you can use the `Ping` button on the router page. It will try to connect to the router and disconnect afterwards. If it succeed, it means NAPALM will be operational for other tasks. The `Poll Sessions` button only retrieve the status of sessions that are in Peering Manager (it does not import them).

**Reply by daverdzx** [reply]
Hello Guillaume, thank you so much for the clarification.  Gonna try the import of sessions over IXPs functionality.

Regards from Flo Networks team, have a great day!
